modules/services/wikidata.js

```diff
@@ -32,9 +32,11 @@ export default {
             search: query,
             type: 'item',
             // the language to search
-            language: lang,
+            // do not include the country-code suffix
+            language: lang.split('-')[0],
```

Comment:
> I'm not sure if simply stripping out the -XX country code suffix is the right solution, but this is an incredible QOL improvement

Unfortunately, this would regress important behavior in several languages, such as Serbian and Chinese, where the country or script would be quite significant. Even in English, Wikibase supports `en-ca`, `en-gb`, and `en-us` but not `en-nz` for some reason. [This API request](https://www.wikidata.org/wiki/Special:ApiSandbox#action=query&format=json&meta=wbcontentlanguages&formatversion=2) returns the list of languages supported for labels in Wikibase.

Considering that NZ English has some terms that differ from both British and American English, it may make sense to request it as a new Wikibase language on Phabricator, where a contributor recently [worked out how to add new languages](https://phabricator.wikimedia.org/T335857#8854666).

Actually, MediaWiki requires the language codes to be fully lowercase, not with uppercase regions as is the norm. iD has a workaround for American English that appears to be unnecessary:

https://github.com/openstreetmap/iD/blob/bd98ff904bb9498ae60fd32548cd2bbcf244bb53/modules/services/wikidata.js#L90-L94

Reply:
thanks, that's really helpful!

I had a look thru [all the languages that iD supports](https://github.com/openstreetmap/iD/tree/develop/dist/locales), and of those, this issue seems to affect `en-NZ`, `en-AU`, `en-IE`, `en-IN`, `nl-NL`, `he-IL`, `fa-IR`, `fr-FR`, and `ar-AA`

I'm a bit confused though, I see [this request to support `en-IN`](https://phabricator.wikimedia.org/T212313) and [this one for `en-AU`](https://phabricator.wikimedia.org/T286862) appear to be "resolved", but neither locale is supported by the API...[^1]

[^1]: I've been testing this by opening [this URL](https://www.wikidata.org/w/api.php?action=wbsearchentities&format=json&formatversion=2&language=en-au&limit=10&origin=*&search=as&type=item&uselang=en-nz) in a new tab and tweaking `language=*`